mlflow/gateway/config.py

```diff
@@ -72,26 +83,59 @@ class RouteType(str, Enum):
 class CohereConfig(ConfigModel):
     cohere_api_key: str
 
-    @validator("cohere_api_key", pre=True)
-    def validate_cohere_api_key(cls, value):
-        return _resolve_api_key_from_input(value)
+    if IS_PYDANTIC_V2:
+        from pydantic import field_validator as _field_validator
+
+        @_field_validator("cohere_api_key", mode="before")
+        @classmethod
+        def validate_cohere_api_key(cls, value):
+            return _resolve_api_key_from_input(value)
+
+    else:
+        from pydantic import validator as _validator
+
+        @_validator("cohere_api_key", pre=True)
+        def validate_cohere_api_key(cls, value):
+            return _resolve_api_key_from_input(value)
```

Comment:
What about defining a new decorator to avoid repeating this if-else logic all over the place?

For example,
```
def pydantic_field_validator(field_name: str):
    def decorator(func: Callable) -> Callable:
        if IS_PYDANTIC_V2:
            from pydantic import field_validator
            return field_validator(field_name, mode="before")(func)
        else:
            from pydantic import validator
            return validator(field_name, pre=True)(func)

    return decorator 
```

then we can do

```
    @pydantic_field_validator("cohere_api_key")
    def validate_cohere_api_key(cls, value):
        return _resolve_api_key_from_input(value)
```